src/rgw/rgw_keystone.cc

```diff
@@ -152,6 +152,88 @@ KeystoneApiVersion KeystoneService::get_api_version()
   }
 }
 
+int RGWSwift::get_keystone_url(CephContext * const cct,
+                               std::string& url)
+{
+  // FIXME: it seems we don't need RGWGetRevokedToken here
+  bufferlist bl;
+  RGWGetRevokedTokens req(cct, &bl);
+
+  url = cct->_conf->rgw_keystone_url;
```

Comment:
This method exports potentially slightly modified (the trailing slash) value of   `cct->_conf->rgw_keystone_url` to a caller. I doubt that `RGWGetRevokedTokens` alters `cct->_conf`.